### ISSUE TITLE
Globbing in `[ .. ]` should not suggest `[[ .. ]]` for POSIX Shell (`sh`)

### ISSUE THREAD

**Issue by ingydotnet** [koalaman/shellcheck#1562]
#### For bugs
- SC2081
- My shellcheck version: 0.6.0 and online 
- [x ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=sh
foo=$(bar)
[ "$foo" = bar* ] && baz
```

#### Here's what shellcheck currently says:
```
           ^-- SC2081: [ .. ] can't match globs. Use [[ .. ]] or case statement.
```

#### Here's what I wanted or expected to see:

The `Use [[ .. ]] or case statement.` part is wrong, since POSIX sh doesn't support `[[ .. ]]`.

The wiki says "In bash/ksh, you can instead use [[ .. ]] which supports this behavior.". shellcheck shouldn't suggest that for shells that don't support it.

**Comment by koalaman** [comment]
Indeed. Thanks!

**Comment by cfmDeveloper** [comment]
What should we use then, because i'm now running into this issue in a script.
